Checking for Correlation between Year and Fastest Time

In [15]:
import pandas as pd

import statsmodels.api as sm

row_time = pd.read_csv('Data Compilation - Sheet11.csv')

model = sm.OLS(row_time['Average Time'],sm.add_constant(row_time['Year']))

results = model.fit()

results.summary()

c:\Users\jaspe\Documents\GitHub\Oxbridge-CamTrad-Data-Science-2022\venv\lib\site-packages\scipy\stats\_stats_py.py:1769: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=8
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           Average Time   R-squared:                       0.123
Model:                            OLS   Adj. R-squared:                 -0.023
Method:                 Least Squares   F-statistic:                    0.8395
Date:                Thu, 04 Aug 2022   Prob (F-statistic):              0.395
Time:                        10:48:37   Log-Likelihood:                -26.924
No. Observations:                   8   AIC:                             57.85
Df Residuals:                       6   BIC:                             58.01
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        346.0407      6.302     54.910      0.000     330.620     361.461
Year           1.1435      1.248      0.916      0.395      -1.910       4.197
==============================================================================
Omnibus:                        2.826   Durbin-Watson:                   2.958
Prob(Omnibus):                  0.243   Jarque-Bera (JB):                0.978
Skew:                           0.349   Prob(JB):                        0.613
Kurtosis:                       1.436   Cond. No.                         11.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Interpretation of above results suggest negligible correlation between year and average time. That is, it is unlikely for there to exist a trned in this time series. This can be seen with $P>|t| = 0.395$ which indicates very little confidence in any correlation.